android/src/main/java/dog/craftz/sqlite_2/RNSqlite2Module.java

```diff
@@ -110,15 +110,49 @@ public void run() {
   }
 
   // do a update/delete/insert operation
-  private SQLitePLuginResult doUpdateInBackgroundAndPossiblyThrow(String sql, String[] bindArgs,
+  private SQLitePLuginResult doUpdateInBackgroundAndPossiblyThrow(String sql, ReadableArray queryArgs,
                                                                   SQLiteDatabase db) {
     debug("\"run\" query: %s", sql);
     SQLiteStatement statement = null;
     try {
       statement = db.compileStatement(sql);
       debug("compiled statement");
-      if (bindArgs != null) {
-        statement.bindAllArgsAsStrings(bindArgs);
+      // Bind all of the arguments
+      for (int i = 0; i < queryArgs.size(); i++) {
+        ReadableType type = queryArgs.getType(i);
+        switch (type) {
+          case Null:
+            statement.bindNull(i + 1);
+            break;
+
+          case Boolean:
+            final long b;
+            if (queryArgs.getBoolean(i)) {
+              b = 1;
+            } else {
+              b = 0;
+            }
+            statement.bindLong(i + 1, b);
+            break;
+
+          case Number:
+            // Potential edge-case
+            // This could be meant to be a floating point number
+            // or a long integer.
+            // JavaScript's Number type represents both as a single thing.
+            final double f = queryArgs.getDouble(i);
+            statement.bindDouble(i + 1, f);
+            break;
+
+          case String:
+            statement.bindString(i + 1, queryArgs.getString(i));
+            break;
+
+          default:
+            // Treat all other values as a string.
+            statement.bindString(i + 1, queryArgs.getString(i));
```

Comment:
I think it's better to throw an `IllegalArgumentException` instead of converting data into string since `Map` and `Array` are unexpected value types.
In this case the result is always unknown.
I'd prefer to be defensive for writing operations to prevent databases from being broken.